dvc/remote/gdrive.py

```diff
@@ -195,10 +207,35 @@ def drive(self):
 
             # Pass non existent settings path to force DEFAULT_SETTINGS loading
             gauth = GoogleAuth(settings_file="")
-            gauth.CommandLineAuth()
 
-            if os.getenv(RemoteGDrive.GDRIVE_USER_CREDENTIALS_DATA):
-                os.remove(self.gdrive_user_credentials_path)
+            try:
+                gauth.CommandLineAuth()
+            except RefreshError as e:
+                raise GDriveAccessTokenRefreshError(
+                    msg="Google Drive's access token " "refreshment is failed",
+                    cause=str(e),
+                )
+            except KeyError as e:
+                raise GDriveMissedCredentialKeyError(
+                    msg='Google Drive\'s user credentials file "{}" '
+                    "misses value for key {}".format(
+                        self.gdrive_user_credentials_path, str(e)
+                    )
+                )
+            except AuthenticationError as e:
+                raise DvcException(
+                    "Google Drive authentication process is failed",
+                    cause=str(e),
+                )
+            except Exception as e:
+                raise DvcException(
+                    "Google Drive authentication process is failed "
+                    "with exception of type: {}.".format(type(e).__name__),
```

Comment:
Why do we need exception type when we specify the cause anyways?

Reply:
The idea was to print as much more helpful information as possible to identify the source of the exception (`cause` seems doesn't contain original exception type anymore and message might be not informative enough). In case if users will start face auth exceptions which are not covered with additional clarification for the moment, it should be easier for us to repro the condition and add new `except` block for such exceptions. If it is not so helpful as I thought, I can remove it.

Reply:
@MaxRis Full traceback will be printed in verbose mode (-v) anyways,  including the type.